cmd/testscript/config.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"clickhouse-playground/cmd/testscript/processor"
+	gconfig "github.com/gookit/config/v2"
+	gyaml "github.com/gookit/config/v2/yaml"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"os"
+	"time"
+)
+
+const DefaultConfigPath = "test_script_config.yml"
+const DefaultRunsDataPath = "runs_data.yml"
+
+type Config struct {
+	Playground PlaygroundSettings `mapstructure:"playground"`
+	TestScript TestScript         `mapstructure:"test_script"`
+	AWS        AWS                `mapstructure:"aws"`
+}
+
+type PlaygroundSettings struct {
+	BaseURL string `mapstructure:"base_url"`
+}
+
+type TestScript struct {
+	Mode         processor.Mode `mapstructure:"mode"`
+	RunsDataPath string         `mapstructure:"runs_data_path"`
+	ResultPath   *string        `mapstructure:"result_path"`
+	DefaultQuery *string        `mapstructure:"default_query"`
+}
+
+type AWS struct {
+	AccessKeyID        string `mapstructure:"access_key_id"`
+	SecretAccessKey    string `mapstructure:"secret_access_key"`
+	Region             string `mapstructure:"region"`
+	QueryRunsTableName string `mapstructure:"query_runs_table"`
+}
+
+func LoadConfig() (*Config, error) {
+	path := os.Getenv("TEST_SCRIPT_CONFIG_PATH")
+	if path == "" {
+		path = DefaultConfigPath
+	}
+
+	gconfig.WithOptions(
+		gconfig.ParseEnv,
+		gconfig.Readonly,
+		func(opts *gconfig.Options) {
+			opts.DecoderConfig = &mapstructure.DecoderConfig{
+				TagName:          "mapstructure",
+				WeaklyTypedInput: true,
+				DecodeHook:       mapstructure.StringToTimeHookFunc(time.RFC3339Nano),
+			}
+		},
+	)
+	gconfig.AddDriver(gyaml.Driver)
+
+	err := gconfig.LoadFiles(path)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load config")
+	}
+
+	cfg := new(Config)
+	err = gconfig.BindStruct("", cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "config binding failed")
+	}
+
+	err = cfg.validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid config")
+	}
+
+	return cfg, nil
+}
+
+// validate verifies the loaded config and sets default values for missed fields.
+func (c *Config) validate() error {
+	if c.Playground.BaseURL == "" {
+		return errors.New("missing playground base url")
+	}
+
+	if c.TestScript.DefaultQuery != nil && *c.TestScript.DefaultQuery == "" {
+		return errors.New("Default query cannot be empty")
+	}
+
+	if c.TestScript.RunsDataPath == "" {
+		c.TestScript.RunsDataPath = DefaultRunsDataPath
+	}
+
+	if c.TestScript.ResultPath == nil {
+		resultPath := "result.csv"
```

Comment:
IMO, it's better for readability to declare all the default values in one place

Reply:
Yes, for sure. Guess I just missed it